### PR TITLE
[CR] Professions and mod update

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
I once again made some changes to scenarios and professions based on my gameplay experience, not my play style, but experience with the professions in game due to their niche, This changes are made to core and to the More professions mod now know as the Professions and Scenarios Mod. 

Moved some professions  from core to the mod that cluttered the core's profession screen that did not contributed to unique gameplay as per @mugling 's suggestion.  

One change in particular might be a bit controversial for @Coolthulhu with the change of the bionic_prepper.

Full change log in respective commit's description.

[CR] since changes might controversial and changes affect all playerbase.
[WIP] Since waiting for review and fix/change requests by devs.

This is not intended to be the final draft and any other changes or profession/scenarios that need to be moved to the mod, feel free to let me know.

P.S: Sorry that the description for the core professions commit came out weird.

### CODE REVIEW

**Comment by Noctifer-de-Mortem** [comment]
Here is the full changelog since it came out bugged in commit description.
[Full Changelog.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/835630/Full.Changelog.txt)

**Comment by kolsurma** [comment]
bionic_prepper is a bit too good for the points now. I guess it'd be a good candidate for the proposed "[easy mode](http://smf.cataclysmdda.com/index.php?topic=14254.0)"

**Comment by Noctifer-de-Mortem** [comment]
I could wilderness lock it as well and/or raise its point beyond 8. Reason no profession so far has been raise to 8 is so player does  not have to sacrifice on traits or if they do it can be for stats.

**Comment by Coolthulhu** [comment]
* Bionic assassin at 8 points can be considered, since he spawns with the best melee weapon in game, but that's a strong nerf right there, to a profession that didn't see much use
* Strong nerf to bionic monster, but it makes sense since bio claws and metabolic at 0 points was rather strong, even with leaky

But the prepper.
Prepper at 8 points is pretty useless. At 8 points, you need to sacrifice trait or stat points. At 6 points you can just go infected. The new bionics are low tier and extra skills are of the ones you can easily grind up (except the first aid) meaning it's still only really about torsion+tools, with everything else being just padding.
And without welding, the tools are mostly just about a toolbox, which isn't a rare drop.
That's a serious nerf to a class that stops shining shortly after start.

**Comment by Noctifer-de-Mortem** [comment]
*Bionic Prepper: 8 points might be too high, mostly raised due to additional bionics.
*Bionic monster: while a nerf, it also allows this class to be used stealthily if you want to do that with the only cost still being performance.
*Bionic Prepper: This is why the commercial cyborg has the toolset,welding googles bionic, hydraulic muscles (to move heavy vehicle objects) and skills for crafting and mechanics. This class is supposed to be the crafter now. I can change the battery bionic to torsion rachet so that it can fully be as the prepper once was. 
Now move this profession to the wild, it has water, means of fire, cooking and crafting with only ingredients to worry about. This why why I also suggest to lock it to wilderness scenarios. 
I can see about changing the skill set, point cost and items included. Adding the welding google back would be something I would be reluctant too, adding welding googles could be a compromise since he is a prepper.

**Comment by kolsurma** [comment]
If they're going to start in the wilderness prepper should probably have some sort of tool/weapon, entrenching tool or hatchet or something like that. Also a small amount of food like a couple pieces of jerky. 

Taking the noisemaker cbm out of bionic monster is appreciated, that one was truly horrible.

**Comment by Coolthulhu** [comment]
> Now move this profession to the wild, it has water, means of fire, cooking and crafting with only ingredients to worry about.

The problem is, it is incredibly expensive and doesn't have much going for it in the wild that wouldn't be available cheaper.
Tools are mostly useful for electricians, mechanicians and chemists, everyone else would be fine with a rambo knife/multitool/toolbox.
Its only combat option is bio_lighter, which is useless for hunting, which is the only food source in winter.
Due to lack of non-fire combat options, it can't even safely set up a house, because any attempt at defense is also a risk of destroying the house.
The bio_lighter is only useful as combat option here, because the tools handle cooking. And even when trying to keep bio power, it's cheaper to start the fire with a fire drill or lighter than the mini-flamethrower.

Overall it doesn't even handle the survival all that well - it needs extra points spent on combat to become viable as survival profession.

**Comment by Noctifer-de-Mortem** [comment]
I agree that this needs a combat option for it to atleast hunt but should I add a weapon (probably ranged) and/or lower the cost to make it more customizable? Adding higher skills to encourage any extra points be used in combat?

As for the toolset, it is intergrated meaning it does not compete with volume for storage space (important if not setting a permant base), but it does compete for bionic slot space (might be erelevant point), and it also doubles as a cooking device. Crafting in the wild is also important, having a toolset to make more complex tools to dig, chop trees or make other equipment is very useful.

The biolighter is meant as a very cheap refillable lighter. Other fire options are cheaper sure, but will eventually run out. The player can craft many firedrills to cover their fire needs or stock up with lighters/matchbooks but if they don't have any other source of fire, the bio ligher can come in handy.

**Comment by Noctifer-de-Mortem** [comment]
Took feed back and applied it hopefully satisfactory. Change log in commit description or read it in the file attached.
[Feedback_changelog.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/836337/Feedback_changelog.txt)

File: data/json/professions.json [hunk 1]
```diff
@@ -117,52 +117,88 @@
...
-                    "canteen"
-                ],
-                "male": [
-                    "boxer_shorts"
+                    "sewing_kit",
+                    "e_tool",
+                    "hatchet",
+                    "mess_kit",
+					"rollmat",
+					"tent_kit"
```
Comment (Coolthulhu) [inline data/json/professions.json]: Tabs

**Comment by Coolthulhu** [comment]
The prepper is a horrible candidate for 8 point profession because:
* Only torsion and bio_tools will matter beyond very early game (climate will only start mattering again in ice labs). Everything else this class offers becomes obsolete fast.
* Lack of points to spare means no customization, which is especially important to those who have no access to books
* Survival (as in - food, water and shelter) is only really an early game concern, which means that the whole class is already bad at everything after early game
* Wilderness necessitates some melee combat skills because no cotton for armor and no punching bags means training up is hard, while wild animals (zombears) are pretty tough for a weak character

You should just rebalance it to fit the 6 points and treat it as a hard limit for any class without really rare bionics or items. At least for scenarios that don't allow starting with more points than that (assuming multi pool of course).

**Comment by Noctifer-de-Mortem** [comment]
6 points it is then. Is the hatchet an acceptable wilderness melee weapon or should I go with something else? A I choose a hatchet since it doubles as a crude axe.

**Comment by Coolthulhu** [comment]
I always pick ice axe over hatchet in the game, but hatchet is still pretty good.
I don't think it actually has the ability to chop wood. At least it didn't have it last time I checked.

**Comment by Coolthulhu** [comment]
Tabs

**Comment by Coolthulhu** [comment]
Machete is a solid early weapon.
Looks pretty solid now. Not exceptional, but a smooth start with some midgame power. Requires taking infected for optimal start, but that machete and combat abilities should make up for it.

**Comment by Noctifer-de-Mortem** [comment]
Should I add it to the infected scenario? If so, what about evacuee or missed scenarios? Should the SCEN_ONLY flag be dropped?

**Comment by Coolthulhu** [comment]
Should be available to infected and sheltered. No reason to keep it limited to those, though.

**Comment by Noctifer-de-Mortem** [comment]
Infected, sheltered, wilderness sounds good. I supposed the SCEN_ONLY flag can be removed.

**Comment by Noctifer-de-Mortem** [comment]
Removed [WIP] since not other changes are being proposed, only fixes.

File: data/mods/more_classes_scenarios/cs_scenarios.json [hunk 2]
```diff
@@ -30,13 +30,41 @@
...
+    "name": "Ambush",
+    "description": "It is the winter after zero hour, cold and weary you ducked into a small cabin for the night.  The next morning, you awoke to the sound of lots of movement in the woods.",
     "copy-from" : "ambushed",
     "extend": { "professions" : [ "wasteland_cyborg" ] }
   },
   {
     "type" : "scenario",
     "ident" : "heli_crash",
+    "name": "Helicopter Crash",
+    "description": "You've crashed your precious helicopter, it'll never fly again.  Monsters don't bother you, life without your Loach is worthless anyway.",
```
Comment (Coolthulhu) [inline data/mods/more_classes_scenarios/cs_scenarios.json]: Is copying the description and name necessary?

File: data/mods/more_classes_scenarios/cs_classes.json [hunk 3]
```diff
@@ -260,5 +260,219 @@
...
+                "football"
+            ],
+            "male": [
+                "briefs"
+            ],
+            "female": [
+                "sports_bra",
+                "panties"
+            ]
+        }
```
Comment (Coolthulhu) [inline data/mods/more_classes_scenarios/cs_classes.json]: Now that multi pool is the default mode, this dude could use some extra bonuses or -1 cost.
Survivor starts with warm clothing, knife, fire and water, football player only has bad armor.

File: data/mods/more_classes_scenarios/cs_classes.json [hunk 4]
```diff
@@ -260,5 +260,219 @@
...
+                "baseball"
+            ],
+            "male": [
+                "briefs"
+            ],
+            "female": [
+                "sports_bra",
+                "panties"
+            ]
+        }
```
Comment (Coolthulhu) [inline data/mods/more_classes_scenarios/cs_classes.json]: Bat is a good start, but 1 point in melee and/or bashing would make it more sensible.

File: data/json/professions.json [hunk 5]
```diff
@@ -2601,30 +2576,17 @@
             "bio_blade",
             "bio_shock",
             "bio_carbon",
+            "bio_adrenaline",
+            "bio_night_vision",
```
Comment (Coolthulhu) [inline data/json/professions.json]: Got the extra bionics, but no longer has the extra cost.

File: data/mods/more_classes_scenarios/cs_scenarios.json (hunk 2 above)
Comment (Noctifer-de-Mortem) [inline data/mods/more_classes_scenarios/cs_scenarios.json]: Apparently. if I don't it throws errors at launch.

**Comment by Noctifer-de-Mortem** [comment]
Astyle regression? I don't follow. What do I need to do to pass the failed check?

**Comment by Coolthulhu** [comment]
If the astyle problem is in a file you didn't change, the bug is in main repo and not caused by you.

**Comment by Noctifer-de-Mortem** [comment]
Ah, ok, thanks!

**Comment by kolsurma** [comment]
@Coolthulhu I think you introduced the astyle in #20549

**Comment by Coolthulhu** [comment]
Bio monster should lose the noise cancelling headgear if he no longer has shrieker faulty bionic.

**Pull request merged**